### PR TITLE
Add docker-compose based workflow, generate production deploy container

### PR DESCRIPTION
**Pull request by msheiny**
Things left to do:
- [x] Fix CI so tests run for dev and CI environment
- [x] Expand documentation
- [x] Fix some DRY issues between dev + prod-like environment

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+version: '3'
+networks:
+  app:
+services:
+  stn_postgresql:
```
Comment (harrislapiroff) [inline docker-compose.yaml]: I would actually love to have these container names unprefixed. If I remember correctly how `docker-compose` works, one of the advantages of it is that it manages container name uniqueness globally as long as you have unique names within a particular compose setup. That way I can do things like ```docker-compose exec django ./manage.py makemigrations app_name --empty``` without having to remember the prefix for the particular application I'm working on.

**Comment by msheiny** [comment]
@harrislapiroff READY FOR RE-REVIEW. Got `b82a16d` passing CI and the last commit was just a set of documentation changes soooo i dont expect it to fail but it takes like 25 min :|

**Comment by conorsch** [comment]
Solid progress here. There are a few outstanding issues in the changes presented that block merge. To wit:

* [ ] Apt dependencies shouldn't be version-locked
* [ ] Makefile help target isn't parsing new targets
* [ ] The wait-for-it script lacks license porting; can we use a simpler approach, e.g. nc?
* [ ] The entrypoint script ignores errors; port to bash and use `set -e`

Once those concerns are addressed, we should be good to go.

**Comment by conorsch** [comment]
See [CI build failure](https://circleci.com/gh/freedomofpress/securethenews/127) demonstrating the problems with version-locking the apt dependencies:

```
ERROR: unsatisfiable constraints:
  curl-7.60.0-r0:
    breaks: world[curl=7.59.0-r0]
ERROR: Service 'django' failed to build: The command '/bin/sh -c apk add --no-cache 
```

Let's rip that out.

**Comment by conorsch** [comment]
CI has been running for an hour on https://circleci.com/gh/freedomofpress/securethenews/129 :grimacing: We're likely gonna have to tighten up the caching a bit here. While working on this PR, noticed that `make dev-go` doesn't make use of the cache frequently—so there's some finetuning yet to do.

**Comment by msheiny** [comment]
> CI has been running for an hour on https://circleci.com/gh/freedomofpress/securethenews/129 grimacing We're likely gonna have to tighten up the caching a bit here. 

So this ended up waiting for 5 hours. This wasnt a caching issue... django fell over. This was why the wait-for-it.sh script was introduced. So django will wait for postgres before starting up. So you took that script out and now the logic replaced with isnt working :( Its kewl tho! we can either fix that OR we can try to add that logic into Django itself.... @harrislapiroff do you know if there is a postgres retry setting?

```
Wait for environemnt to come up completely
...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................Step was canceled
```

File: docker/django-start.sh [hunk 2]
```diff
@@ -13,21 +15,23 @@ wait_for_node() {
 }
 
 wait_for_postgres() {
-    echo Waiting for postgres to start..
-    wait-for-it.sh -t 120 -h ${DJANGO_DB_HOST} -p ${DJANGO_DB_PORT}
+    echo "Waiting for postgres to start..."
+    until nc -z "${DJANGO_DB_HOST}" -p "${DJANGO_DB_PORT}"
```
[the remark this answers is not in the document]
Comment (msheiny) [reply docker/django-start.sh]: huh nope... that syntax is... well ... let me try some stuff. might have mispoken here :)

**Comment by conorsch** [comment]
Great fixes, @msheiny! Thanks for cleaning up my dust piles with the wait-for-it/nc changes. Quite pleased with the progress here. Surely we'll find additional touch-ups while working on the staging integration, but for now we've got a solid foundation to get rolling!

**Pull request merged**